### PR TITLE
Make parity-bitcoin usable for spv validation

### PR DESCRIPTION
**Pull request by alekseysidorov**
- In this PR I extract from the crate `db` to the new `storage` crate all traits that is not specifically related to the concrete implementation of database. So I now can use `verification` crate without `rocksdb` dependency.

- Also I modify the `HeaderAcceptor` so that can be constructed without the private `BlockDeployments`.

[clabot:check]

### CODE REVIEW

**Comment by svyatonik** [comment]
Excellent idea, thanks for your efforts!

File: db/Cargo.toml [hunk 1]
```diff
@@ -7,12 +7,13 @@ authors = ["Parity Technologies <admin@parity.io>"]
 rocksdb = { git = "https://github.com/ethcore/rust-rocksdb" }
 elastic-array = "0.6"
 parking_lot = "0.4"
-log = "0.4"
+log = "0.3"
```
Comment (svyatonik) [inline db/Cargo.toml]: Is this intentional or not?

Comment (alekseysidorov) [reply db/Cargo.toml]: Oh, it is my mistake.

File: storage/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	DatabaseError(String),
+	/// Invalid block
+	CannotCanonize,
+	/// Uknown parent
+	UnknownParent,
+	/// Ancient fork
+	AncientFork,
+}
+
+impl From<Error> for String {
```
Comment (debris) [inline storage/src/error.rs]: this is not an idiomatic way of printing errors. It's better to `impl fmt::Debug for Error` and `impl fmt::Display for Error`

**Comment by debris** [comment]
nice!

File: storage/src/error.rs (hunk 2 above)
Comment (svyatonik) [reply storage/src/error.rs]: I guess it is a copy of `db/src/error.rs`. Do we still need `db/src/error.rs`? Can't we just use `storage::Error` everywhere? If so, could you please use `git mv` for these files?

Comment (alekseysidorov) [reply storage/src/error.rs]: @debris 
What do you think about usage https://crates.io/crates/display_derive?
@svyatonik I got it.

Comment (debris) [reply storage/src/error.rs]: @alekseysidorov we can use it ;>

Comment (alekseysidorov) [reply storage/src/error.rs]: @debris Ready.

But, in my opinion, to integrate a `failure` crate is a more idiomatic and complete solution.

**Comment by rphmeier** [comment]
(note: needs CLA bot signoff before merge!)

**Comment by alekseysidorov** [comment]
[clabot:check]

**Pull request merged**